lib/HunkView.ml

```diff
@@ -0,0 +1,178 @@
+          W.string
+            ~attr:Notty.A.(fg green)
+            (Printf.sprintf "   %2d + %s" (their_line_num + 1) line) )
+    | `Mine line ->
+        ( (mine_line_num + 1, their_line_num),
+          W.string
+            ~attr:Notty.A.(fg red)
+            (Printf.sprintf "%2d    - %s" (mine_line_num + 1) line) )
+  in
+  (* _ is the final state of the lines, if needed later *)
+  let _, lines_ui =
+    List.fold_left_map update_lines initial_line_nums hunk.Patch.lines
+  in
+  let lines_ui_vcat = Ui.vcat lines_ui in
+
+  Ui.vcat [ ui_hunk_summary hunk; lines_ui_vcat ]
+
+let current_hunks z_patches : ui Lwd.t =
+  let$ z = Lwd.get z_patches in
+  let p = Zipper.get_focus z in
+  let hunks = List.map ui_unified_diff p.Patch.hunks in
+  Ui.vcat hunks
+
+(** Side by side diff view implementation **)
+
+type view_mode = SideBySide | Normal
+
+let view_mode = Lwd.var Normal
+
+let toggle_view_mode () =
+  match Lwd.peek view_mode with
+  | Normal -> Lwd.set view_mode SideBySide
+  | SideBySide -> Lwd.set view_mode Normal
+
+type line = Change of string | Common of string | Empty
+
+let rec split_and_align_hunk hunks mine_acc their_acc =
+  let rec buffer_changes temp_mine temp_their hunks =
+    match hunks with
+    | [] -> (temp_mine, temp_their, hunks)
+    | `Common _ :: _ -> (temp_mine, temp_their, hunks)
+    | `Mine line :: t -> buffer_changes (Change line :: temp_mine) temp_their t
+    | `Their line :: t -> buffer_changes temp_mine (Change line :: temp_their) t
+  in
+  let append_balanced temp_mine temp_their =
+    let mine_len = List.length temp_mine in
+    let their_len = List.length temp_their in
+    let fill_empty n = List.init n (fun _ -> Empty) in
+
+    let diff = mine_len - their_len in
+    let empty_list = fill_empty (abs diff) in
+
+    if diff > 0 then (temp_mine, empty_list @ temp_their)
+    else if diff < 0 then (empty_list @ temp_mine, temp_their)
+    else (temp_mine, temp_their)
+  in
+
+  match hunks with
+  | [] -> (List.rev mine_acc, List.rev their_acc)
+  | _ -> (
+      let temp_mine, temp_their, remaining_hunks = buffer_changes [] [] hunks in
+      let balanced_mine, balanced_their =
+        append_balanced temp_mine temp_their
+      in
+      let updated_mine_acc = balanced_mine @ mine_acc in
+      let updated_their_acc = balanced_their @ their_acc in
+      match remaining_hunks with
+      | `Common line :: t ->
+          let common_mine_acc = Common line :: updated_mine_acc in
+          let common_their_acc = Common line :: updated_their_acc in
+          split_and_align_hunk t common_mine_acc common_their_acc
+      | _ ->
+          split_and_align_hunk remaining_hunks updated_mine_acc
+            updated_their_acc)
+
+let lines_with_numbers lines attr_change prefix =
+  let line_num = ref 0 in
+  List.fold_left
+    (fun acc line ->
+      match line with
+      | Common s ->
+          incr line_num;
+          let content = Printf.sprintf "%3d   %s" !line_num s in
```

Comment:
Now all line numbers are considered to have 3 digits: that's a lot! :)

But no need to change anything in this PR: you have opened #66 anyway.